File: boj/10799.cpp
#include <iostream>
#include <stack>

using namespace std;

stack<int> s;

int main(void) {
    ios_base::sync_with_stdio(false);
    string str; cin >> str;
    int len = str.length();
    int ans = 0;
    for (int i = 0; i < len; ++i) {
        if (str[i] == '(') {
            s.push(i);
        } else {
            if (s.top() == (i - 1)) {
                s.pop();
                ans += s.size();
            } else {
                s.pop();
                ++ans;
            }
        }
    }
    cout << ans << "\n";
    return 0;
}

File: boj/10799.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Stack;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        Stack<Integer> s = new Stack<>();
        String str = br.readLine();
        int len = str.length();
        int ans = 0;
        for (int i = 0; i < len; ++i) {
            if (str.charAt(i) == '(') {
                s.push(i);
            } else {
                if (s.peek() == (i - 1)) {
                    s.pop();
                    ans += s.size();
                } else {
                    s.pop();
                    ++ans;
                }
            }
        }
        System.out.println(ans);
    }
}
